docs/by-example/sqlalchemy/advanced-querying.rst

```diff
@@ -0,0 +1,267 @@
+    >>> from uuid import uuid4
+
+Establish a connection to the database, see also :ref:`sa:engines_toplevel`
+and :ref:`connect`:
+
+    >>> engine = sa.create_engine(f"crate://{crate_host}")
+    >>> connection = engine.connect()
+
+Create an SQLAlchemy :doc:`Session <sa:orm/session_basics>`:
+
+    >>> session = sessionmaker(bind=engine)()
+    >>> Base = declarative_base()
+
+
+Introduction to fulltext indexes
+================================
+
+:ref:`crate-reference:fulltext-indices` take the contents of one or more fields
+and split it up into tokens that are used for fulltext-search. The transformation
+from a text to separate tokens is done by an analyzer. In order to create
+fulltext search queries a :ref:`fulltext index with an analyzer
+<crate-reference:sql_ddl_index_fulltext>` must be defined for the related columns.
+
+In order to support fulltext query operations, the CrateDB SQLAlchemy dialect
+provides the :ref:`crate-reference:predicates_match` through its ``match`` function.
+
+For exercising those features, let's define a schema using SQLAlchemy's
+:ref:`sa:orm_declarative_mapping`:
+
+    >>> def gen_key():
+    ...     return str(uuid4())
+
+    >>> class Character(Base):
+    ...     __tablename__ = 'characters'
+    ...     id = sa.Column(sa.String, primary_key=True, default=gen_key)
+    ...     name = sa.Column(sa.String)
+    ...     quote = sa.Column(sa.String)
+    ...     name_ft = sa.Column(sa.String)
+    ...     quote_ft = sa.Column(sa.String)
+    ...     __mapper_args__ = {
+    ...         'exclude_properties': ['name_ft', 'quote_ft']
+    ...     }
+
+Please note that the schema defined above is provisioned to the database using
+the following SQL DDL statement.
+
+.. code-block:: sql
+
+    CREATE TABLE characters (
+        id STRING PRIMARY KEY,
+        name STRING,
+        quote STRING,
+        INDEX name_ft USING fulltext(name) WITH (analyzer = 'english'),
+        INDEX quote_ft USING fulltext(quote) WITH (analyzer = 'english')
+    )
+
+.. note::
+
+    Currently, it is not supported to define a *named index column definition*
+    using a :ref:`crate-reference:sql_ddl_index_fulltext` on behalf of the
+    SQLAlchemy declarative schema.
+
+Let's add two records we use for testing.
+
+    >>> arthur = Character(name='Arthur Dent')
+    >>> arthur.quote = "Let's go somewhere."
+    >>> session.add(arthur)
+
+    >>> trillian = Character(name='Tricia McMillan')
+    >>> trillian.quote = "We're on a space ship Arthur. In space."
+    >>> session.add(trillian)
+
+    >>> session.commit()
+
+After ``INSERT`` statements are submitted to the database, the newly inserted
+records aren't immediately available for retrieval because the index is only
+updated periodically (default: each second). In order to synchronize that,
+refresh the table:
+
+    >>> _ = connection.execute(sa.text("REFRESH TABLE characters"))
+
+
+Aggregates: Counting and grouping
+=================================
```

Comment:
Maybe move this elsewhere - the previous section is "Introduction to fulltext indexes" and the next section is "Fulltext search with MATCH predicate" - having an intermediate aggregation/count/group-by breaks the flow

Reply:
I agree and will try to find a different spot.

Reply:
The section has been moved one slot to the bottom with fe9030016.